datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxAttachService.java

```diff
@@ -0,0 +1,132 @@
+package com.sequenceiq.datalake.service.sdx.attach;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.authorization.service.OwnerAssignmentService;
+import com.sequenceiq.cloudbreak.common.service.TransactionService;
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.repository.SdxClusterRepository;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+
+/**
+ * Provides core logic and functionality for attaching or reattaching an SDX cluster.
+ */
+@Component
```

Comment:
nit: should be `@service` if the class is called service

Reply:
Done but I would appreciate if you quickly explained the difference between a component and a service. I just did this because I saw similar files use component.

Reply:
no major difference, but if you call your class service then it makes sense to use a matching annotation. It's just a classification thing, that's why it's just a small thing, which wouldn't block merging a pr. I wouldn't have asked it to be changed if there wasn't any other issue
https://www.baeldung.com/spring-component-repository-service